### ISSUE TITLE
test_arrayeditor.py::test_arrayeditor_with_inf_array fails because of deprecation warning (Py3.8)

### ISSUE THREAD

**Issue by bnavigator** [spyder-ide/spyder#11885]
(Note that I have slightly amended the test function to print out the recorded warning)

```
abuild@greinerT450s:~/rpmbuild/BUILD/spyder-4.1.1> pytest -vvvv -k test_arrayeditor_with_inf_array
================================================= test session starts =================================================
platform linux -- Python 3.8.2, pytest-4.6.9, py-1.8.1, pluggy-0.13.1 -- /usr/bin/python3
cachedir: .pytest_cache
PyQt5 5.13.2 -- Qt runtime 5.14.1 -- Qt compiled 5.14.1
rootdir: /home/abuild/rpmbuild/BUILD/spyder-4.1.1, inifile: pytest.ini
plugins: qt-3.3.0, xvfb-1.2.0, lazy-fixture-0.6.3, mock-2.0.0, timeout-1.3.4, cov-2.8.1, flaky-3.6.1
collecting ... QStandardPaths: XDG_RUNTIME_DIR not set, defaulting to '/tmp/runtime-abuild'
collected 1094 items / 1093 deselected / 1 selected                                                                   

spyder/plugins/variableexplorer/widgets/tests/test_arrayeditor.py::test_arrayeditor_with_inf_array FAILED       [100%]

====================================================== FAILURES =======================================================
___________________________________________ test_arrayeditor_with_inf_array ___________________________________________

qtbot = <pytestqt.qtbot.QtBot object at 0x7f97018d3430>, recwarn = WarningsRecorder(record=True)

    def test_arrayeditor_with_inf_array(qtbot, recwarn):
        """See: spyder-ide/spyder#8093"""
        arr = np.array([np.inf])
        res = launch_arrayeditor(arr, "inf array")
        for r in recwarn:
            print(r.category)
            print(r.message)
            print(res)
>       assert len(recwarn) == 0
E       assert 2 == 0
E         -2
E         +0

spyder/plugins/variableexplorer/widgets/tests/test_arrayeditor.py:128: AssertionError
------------------------------------------------ Captured stdout call -------------------------------------------------
<class 'DeprecationWarning'>
an integer is required (got type float).  Implicit conversion to integers using __int__ is deprecated, and may be removed in a future version of Python.
[inf]
<class 'DeprecationWarning'>
an integer is required (got type float).  Implicit conversion to integers using __int__ is deprecated, and may be removed in a future version of Python.
[inf]
``'

**Comment by goanpeca** [comment]
Thanks! I will add a fix for this :-) I wrote that test I think.

**Comment by bnavigator** [comment]
I hope you didn't start yet. See my fix in #11899
